packages/ember-glimmer/lib/components/curly-component.js

```diff
@@ -66,6 +67,11 @@ class CurlyComponentManager {
       bucket.classRef = args.named.get('class');
     }
 
+    assert('You cannot use `classNameBindings` on a tag-less component: ' + component.toString(), () => {
```

Comment:
It doesn't appear that there was ever a test for that. I will add the assertions and tests. Any other assertions that need to happen? Events, `classNameBindings`, and `$()` are the only things asserted right now.